llm4crs/agent_plan_first_openai.py

```diff
@@ -52,22 +60,40 @@ def _check_plan_legacy(self, plans: Dict) -> bool:
                 return False
         return True
 
-    def run(self, inputs: str):
+    def run(self, inputs: str, user_profile: UserProfileMemory=None):
```

Comment:
Have you tested whether the original performance (no life-long experience, i.e., when long-term memory is disable and only short-term memory exists) remains good after adding the user profile mechanism?

Reply:
Yes, the introduction of user profile does not affects original performance.